cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -1061,18 +1072,25 @@ private Metadata writeWithHistory(Metadata existing, MetadataEntry entry, Set<In
     return updatedMetadata;
   }
 
+
   /**
    * Returns all the {@link Indexer indexers} that apply to a specified metadata key.
    * @param key the metadata key
+   * @param includeFirstRecordIndexers whether to include indexers which should be used only if this is the first
+   * time a metadata is being stored for the given metadata entity
```

Comment:
looks like what you want to document here is: whether this is the first time the entity is indexed. And name the parameter ```firstTime``` or ```isNewEntity```.

Reply:
Done.